pallets/file-system/src/utils.rs

```diff
@@ -548,8 +548,32 @@ where
             &added_file_key_proof,
         )?;
 
-        // TODO: Check if this is the first file added to the BSP's Forest. If so, initialise
-        // TODO: last block proven by this BSP accordingly.
+        // Check if this is the first file added to the BSP's Forest. If so, initialise last block proven by this BSP.
+        let old_root = expect_or_err!(
+            <T::Providers as shp_traits::ProvidersInterface>::get_root(bsp_id),
+            "Failed to get root for BSP, when it was already checked to be a BSP",
+            Error::<T>::NotABsp
+        );
+        if old_root == Default::default() {
```

Comment:
Ok so `old_root` can be any type implementing Default. And the file system pallet assumes that if the value is default, then it is the first file. 

I was thinking of `Some(root)` instead. So if it is `None`, it means it is the first file.